torch/csrc/dynamo/eval_frame.c

```diff
@@ -786,18 +1066,42 @@ static PyObject* set_eval_frame_py(PyObject* dummy, PyObject* callback) {
   return set_eval_frame(callback, PyThreadState_GET());
 }
 
+static void destroy_cache_entry_wrapper(PyObject* code) {
```

Comment:
Agreed here, especially since you already went through the trouble of defining a PyType for CacheEntryWrapperType. You should just be decref'ing extra at this point.